fe/fe-core/src/main/java/com/starrocks/sql/analyzer/AnalyzerUtils.java

```diff
@@ -155,14 +156,44 @@ public static Function getUdfFunction(ConnectContext session, FunctionName fnNam
             // check SELECT action on any object(table/view/mv) in db
             if (!PrivilegeManager.checkActionInDb(session, dbName, "SELECT")) {
                 throw new StarRocksPlannerException(String.format("Access denied. " +
-                        "Found UDF: %s and need the SELECT action on any object(table/view/mv) in db %s",
+                                "Found UDF: %s and need the SELECT action on any object(table/view/mv) in db %s",
                         fnName, dbName), ErrorType.USER_ERROR);
             }
         }
+        return fn;
+    }
 
-        if (!Config.enable_udf) {
-            throw new StarRocksPlannerException("CBO Optimizer don't support UDF function: " + fnName,
-                    ErrorType.USER_ERROR);
+    private static Function getGlobalUdfFunction(ConnectContext session, FunctionName fnName, Type[] argTypes) {
+        Function search = new Function(fnName, argTypes, Type.INVALID, false);
+        Function fn = session.getGlobalStateMgr().getGlobalFunctionMgr()
+                .getFunction(search, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
+
+        if (fn == null) {
+            return null;
+        }
+
+        if (!GlobalStateMgr.getCurrentState().isUsingNewPrivilege()) {
```

Comment:
Why not check before actually search function

Reply:
Because I guess there are some type conversions or fuzzy matching of function. 

I guess if we use UDF like `aaa(int, int)`,  maybe we don't have exact function like `aaa(int, int)`, but we possibly have function like `aaa(long, long)`. It also works.

```
        Function search = new Function(fnName, argTypes, Type.INVALID, false);
        Function fn = session.getGlobalStateMgr().getGlobalFunctionMgr()
                .getFunction(search, Function.CompareMode.IS_NONSTRICT_SUPERTYPE_OF);
```